pkg/ctl/cmdutils/configfile.go

```diff
@@ -454,18 +454,34 @@ func NewUtilsEnableEndpointAccessLoader(cmd *Cmd, privateAccess, publicAccess bo
 }
 
 // NewUtilsAssociateIAMOIDCProviderLoader will load config or use flags for 'eksctl utils associal-iam-oidc-provider'
-func NewUtilsAssociateIAMOIDCProviderLoader(cmd *Cmd) ClusterConfigLoader {
+func NewUtilsAssociateIAMOIDCProviderLoader(cmd *Cmd, oidcClientIDs []string) ClusterConfigLoader {
 	l := newCommonClusterConfigLoader(cmd)
 
 	l.validateWithoutConfigFile = func() error {
 		l.ClusterConfig.IAM.WithOIDC = api.Enabled()
+		l.ClusterConfig.IAM.OIDCClientIDs = oidcClientIDs
 		return l.validateMetadataWithoutConfigFile()
 	}
 
 	l.validateWithConfigFile = func() error {
 		if api.IsDisabled(l.ClusterConfig.IAM.WithOIDC) {
 			return fmt.Errorf("'iam.withOIDC' is not enabled in %q", l.ClusterConfigFile)
 		}
+
+		oidcClientIDs = append(oidcClientIDs, l.ClusterConfig.IAM.OIDCClientIDs...)
```

Comment:
As mentioned in my comment, we should disallow passing `oidc-client-ids` if a ClusterConfig file is passed (`-f config.yaml`).